### PR TITLE
State not specified for registered sidechain

### LINKED ISSUE #1511: State not specified for registered sidechain
### Actual behavior

`/api/v3/blockchain/apps` endpoint returns response with `state: null,` for a registered sidechain.

```
{
  "data": [
    {
      "name": "enevti",
      "chainID": "04000001",
      "state": null,
      "address": "lskguo9kqnea2zsfo3a6qppozsxsg92nuuma3p7ad",
      "lastCertificateHeight": 2393,
      "lastUpdated": 1676564440
    }
  ],
  "meta": {
    "count": 1,
    "offset": 0,
    "total": 1
  }
}
```

### Expected behavior

Expected `state: 'registered'` in the response.

### Steps to reproduce

- Run Lisk Core node
- Run Lisk sidechain node
- Register the sidechain on the mainchain
- Run Lisk Service against Lisk Core and query `/api/v3/blockchain/apps` endpoint

### Which version(s) does this affect? (Environment, OS, etc...)

Lisk Service v0.7.0-alpha.7

### PR DESCRIPTION
**Pull request by priojeetpriyom**
### What was the problem?
This PR resolves #1511 

### How was it solved?
- [x] Update the method signatures for `applyTransaction` and `revertTransaction` to include `events` as params
- [x] Update transaction processor indexing logic to set correct state
- [x] Update `state` to `status` in the response (Breaking change)

### How was it tested?
- [x] Local

### CODE REVIEW

File: services/blockchain-connector/methods/interoperability.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+const {
+	getChainAccount,
+} = require('../shared/sdk');
+
+const regex = require('../shared/utils/regex');
+
+module.exports = [
+	{
+		name: 'getChainAccount',
+		controller: async ({ chainID }) => getChainAccount({ chainID }),
```
Comment (nagdahimanshu) [inline services/blockchain-connector/methods/interoperability.js:25]: Please update it to make it consistent:
```suggestion
		controller: async ({ chainID }) => getChainAccount(chainID),
```

File: services/blockchain-connector/shared/sdk/interoperability.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+} = require('lisk-service-framework');
+
+const { timeoutMessage, invokeEndpoint } = require('./client');
+
+const logger = Logger();
+
+const getChainAccount = async (params) => {
+	try {
+		const chainAccount = await invokeEndpoint('interoperability_getChainAccount', { chainID: params.chainID });
+		return chainAccount;
```
Comment (nagdahimanshu) [inline services/blockchain-connector/shared/sdk/interoperability.js:28]:
```suggestion
const getChainAccount = async (chainID) => {
	try {
		const chainAccount = await invokeEndpoint('interoperability_getChainAccount', { chainID });
		return chainAccount;
```

File: services/blockchain-connector/shared/sdk/interoperability.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+	} catch (err) {
+		if (err.message.includes(timeoutMessage)) {
+			throw new TimeoutException('Request timed out when calling \'getChainAccount\'.');
+		}
+		logger.warn(`Error returned when invoking 'interoperability_getChainAccount' with chainID: ${params.chainID}.\n${err.stack}`);
```
Comment (nagdahimanshu) [inline services/blockchain-connector/shared/sdk/interoperability.js:33]:
```suggestion
		logger.warn(`Error returned when invoking 'interoperability_getChainAccount' with chainID: ${chainID}.\n${err.stack}`);
```

File: services/blockchain-indexer/shared/dataService/business/interoperability/chainAccount.js [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ *
+ */
+
+const { requestConnector } = require('../../../utils/request');
+
+const getChainAccount = async params => {
```
Comment (nagdahimanshu) [inline services/blockchain-indexer/shared/dataService/business/interoperability/chainAccount.js:19]: May be we can completely remove the method `getChainAccount` from `dataService` layer as it is not been used in `dataService` and we can directly request the connector in indexer wherever it needed. Just a suggestion.

File: services/gateway/sources/version3/mappings/blockchainApp.js [hunk 5]
```diff
@@ -16,7 +16,7 @@
 module.exports = {
 	name: '=,string',
 	chainID: '=,string',
-	state: '=,string',
+	status: '=,string',
```
Comment (nagdahimanshu) [inline services/gateway/sources/version3/mappings/blockchainApp.js:19]: With this change, we also need to update integration test schema at `tests/schemas/api_v3/blockchainAppsSchema.schema.js`

File: services/blockchain-indexer/shared/dataService/business/interoperability/chainAccount.js (hunk 4 above)
Comment (sameersubudhi) [reply services/blockchain-indexer/shared/dataService/business/interoperability/chainAccount.js:19]: It's fine either ways. It probably will be cleaner to directly invoke the connector with `requestConnector` in the `getChainStatus` method in `services/blockchain-indexer/shared/indexer/transactionProcessor/interoperability/registerMainchain.js`.

`Ref: services/blockchain-indexer/shared/indexer/transactionProcessor/index.js`

**Review by sameersubudhi — CHANGES_REQUESTED** [review]
It also seems like we need to adjust the transaction processing logic for `registerMainchain` command. Will let you know the changes offline.

**Comment by sameersubudhi** [comment]
@priojeetpriyom Could you also please the `serviceURL` for `alphanet` in `services/blockchain-indexer/config.js`?
Also, `resolveMainchainServiceURL` seems buggy. Let's discuss the change.

File: services/blockchain-indexer/shared/dataService/business/interoperability/chainAccount.js (hunk 4 above)
Comment (priojeetpriyom) [reply services/blockchain-indexer/shared/dataService/business/interoperability/chainAccount.js:19]: I can understand that we are passing the `getChainAccount` through layers which are not using it. But doing this is keeping all the module related invocations in one place and also keeping (currently most of the) the external calls  under one directory path. For this reason, I would prefer exposing it from the dataservice. Please correct me incase I am headed to a wrong direction guys.

File: services/gateway/apis/http-version3/swagger/definitions/blockchainApps.json [hunk 6]
```diff
@@ -20,10 +20,10 @@
 				"example": "00000000",
 				"description": "Chain ID of the blockchain application."
 			},
-			"state": {
+			"status": {
 				"type": "string",
 				"example": "active",
-				"description": "Current state of the application"
+				"description": "Current status of the application"
```
[the remark this answers is not in the document]
Comment (priojeetpriyom) [reply services/gateway/apis/http-version3/swagger/definitions/blockchainApps.json]: Applied.

File: services/gateway/sources/version3/mappings/blockchainApp.js (hunk 5 above)
Comment (priojeetpriyom) [reply services/gateway/sources/version3/mappings/blockchainApp.js:19]: Applied.

**Review by sameersubudhi — APPROVED** [review]

**Review by nagdahimanshu — APPROVED** [review]
LGTM 👍

**Pull request merged**